frontend/components/organisms/AnswerForm/index.tsx

```diff
@@ -0,0 +1,224 @@
+                        question_id,
+                    },
+                })
+                    .then(() => {
+                        const qlEditor = document.querySelector(
+                            '.quill .ql-editor'
+                        ) as HTMLDivElement
+                        qlEditor.innerHTML = ''
+                        successNotify('Answer Successfully Added!')
+                    })
+                    .catch(() => {
+                        errorNotify('There was an Error!')
+                    })
+
+                setAddAnswerError('')
+                errorElement.classList.remove('error-form-element')
+
+                refetchHandler()
+
+                setTimeout(() => {
+                    setIsDisableSubmit(false)
+                }, 3000)
+            }
+        } else {
+            setAddAnswerError('No answer Input')
+            errorElement.classList.add('error-form-element')
+            setTimeout(() => {
+                setIsDisableSubmit(false)
+            }, 2000)
+        }
+    }
+
+    const onEditSubmit = (data: FormValues) => {
+        setIsDisableSubmit(true)
+        const errorElement = document.querySelector('.ql-container') as HTMLDivElement
+
+        const pattern = /<img[^>]+>/
+        const check_image = pattern.test(data.content)
+        if (
+            data.content.replace(/<(.|\n)*?>/g, '').trim().length === 0 &&
+            check_image === false &&
+            answer.content === null
+        ) {
+            errorElement.classList.add('error-form-element')
+            setAddAnswerError('No answer Input')
+            setTimeout(() => {
+                setIsDisableSubmit(false)
+            }, 2000)
+        } else {
+            const newUpdateAnswer = updateAnswer({
+                variables: {
+                    id: answer.id,
+                    content:
+                        data.content.replace(/<(.|\n)*?>/g, '').trim().length === 0
+                            ? answer.content
+                            : data.content,
+                },
+            })
+                .then(() => {
+                    const qlEditor = document.querySelector('.quill .ql-editor') as HTMLDivElement
+                    qlEditor.innerHTML = ''
+                    successNotify('Answer Successfully Added!')
+
+                    router.push(`/questions/${slug}`)
+                })
+                .catch(() => {
+                    errorNotify('There was an Error!')
+                })
+
+            setAddAnswerError('')
+            errorElement.classList.remove('error-form-element')
+
+            refetchHandler()
+
+            onEdit({ id: null, content: null })
+
+            setTimeout(() => {
+                setIsDisableSubmit(false)
+            }, 3000)
+        }
+    }
+
+    return (
+        <div className="w-full py-3 " id="answer-form">
+            <div className="flex flex-col space-y-2">
+                {answer.id === null ? (
```

Comment:
This could probably be shortened if you have the values for Label, Button Text, value, and onSubmit/onEditSubmit be controlled by state
`<span>{state['onEdit'].label}</span>`
`handleSubmit(state['onEdit'].handleOption)`
`<Button>handleSugmit(state['onEdit'].buttonText<Button>`

There's some examples here
https://github.com/framgia/sph-sun-overflow/commit/6c399f3e24c7bbf55d83db3d0cac8e10926b9b46